File: 0009-Palindrome-Number/0009.java
class Solution {
    public boolean isPalindrome(int x) {
        
        if(x < 0) return false;
        if(x == 0) return true;

        int num = x; 
        long rev = 0; 

        while(num > 0) {
            int digit = num % 10; 
            rev = rev * 10 + digit; 
            num /= 10;
        }

        return x == rev;
    }
}

File: 0009-Palindrome-Number/0009.rs
impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        
        if x < 0 {
        	return false;
        }

        if x == 0 {
        	return true;
        }

        let mut num = x; 
        let mut rev = 0; 

        while num > 0 {
        	let digit = num % 10; 
        	rev = rev * 10 + digit; 
        	num /= 10;
        }

        return x == rev;
    }
}